libraries/core_libs/network/src/tarcap/packets_handlers/pbft_sync_packet_handler.cpp

```diff
@@ -59,6 +53,21 @@ void PbftSyncPacketHandler::process(const PacketData &packet_data, const std::sh
     throw MaliciousPeerException("Unable to parse SyncBlock: " + std::string(e.what()));
   }
 
+  auto sync_period = sync_block.pbft_blk->getPeriod();
+
+  if (!pbft_syncing_state_->updatePeerSyncingPeriod(packet_data.from_node_id_, sync_period)) {
+    std::ostringstream err_msg;
+    err_msg << "PbftSyncPacket received from peer " << packet_data.from_node_id_ << " sending previous period "
+            << sync_period << " PBFT block";
+    throw MaliciousPeerException(err_msg.str());
+  }
+
+  if (pbft_syncing_state_->syncingPeer() != packet_data.from_node_id_) {
```

Comment:
this should be first check before `updatePeerSyncingPeriod` and maybe we should also mark peer as malicious?

Reply:
I think this should not before updatePeerSyncingPeriod

There is an example:
1. A node1 first sync with node2, node2 send PBFT blocks B1, B2, B3 to node1
2. Node1 received B1 and B2. But B3 take a long time to deliver(maybe because network delay or something else caused B3 cannot deliver on time)
3. Node1 switch to sync with node3, node3 send B3, B4 to node1
4. Node1 receive B3, B4 from node3
5. node1 receive B3 from node2
In the case, node3 from node2 is valid, node2 is not malicious.